Assets/Scripts/EnemyController.cs

```diff
@@ -14,87 +15,100 @@ public class EnemyController : MonoBehaviour
+                }
+            }
+        }
+
+
+        public void Move(float move, float stoppingDistance, Vector3 direction, Vector3 playerPosition)
         {
             //only control the enemy if grounded or airControl is turned on
             if (grounded)
             {
                 // Move the character by finding the target velocity
-                Vector3 targetVelocity = new Vector2(move * 5f, rigidBody.velocity.y);
+                Vector3 targetVelocity = new Vector2(move * 5f, rigidBody.velocity.y) * direction;
                 // And then smoothing it out and applying it to the character
-                rigidBody.velocity = Vector3.SmoothDamp(rigidBody.velocity, targetVelocity, ref velocity, movementSmoothing);       
+                rigidBody.velocity = Vector3.SmoothDamp(rigidBody.velocity, targetVelocity, ref velocity, movementSmoothing);
 
-                if (Vector2.Distance(transform.position, playerPosition) < stoppingDistance)
+                if (Vector2.Distance(transform.position, playerPosition) <= stoppingDistance)
                 {
                     rigidBody.velocity = Vector3.zero;
                     anim.SetBool("IsAttacking", true);
                 }
 
                 // If the enemy is moving left and the enemy is facing right...
-                if (move > 0 && facingLeft)
+                if (rigidBody.velocity.x < 0)
                 {
                     // ... flip the enemy.
-                    Flip();
+                    spriteRenderer.flipX = true;
                 }
                 // Otherwise if the enemy is moving the right and the enemy is facing left...
-                else if (move < 0 && !facingLeft)
+                else if (rigidBody.velocity.x > 0)
                 {
                     // ... flip the enemy.
-                    Flip();
+                    spriteRenderer.flipX = false;
                 }
             }
         }
 
-        private void Awake()
+        public void MoveAway(Vector2 distance)
         {
-            rigidBody = GetComponent<Rigidbody2D>();
-
-            if (OnLandEvent == null)
-                OnLandEvent = new UnityEvent();
+            rigidBody.AddForce(distance * 2.0f, ForceMode2D.Impulse);
         }
-
-        private void FixedUpdate()
+        public void Die()
         {
-            bool wasGrounded = grounded;
-            grounded = false;
-
-            // The enemy is grounded if a circlecast to the groundcheck position hits anything designated as ground
-            // This can be done using layers instead but Sample Assets will not overwrite your project settings.
-            Collider2D[] colliders = Physics2D.OverlapCircleAll(groundCheck.position, GroundedRadius, whatIsGround);
-            for (int i = 0; i < colliders.Length; i++)
+            Debug.Log("Enemy has been hit");
+            health.ModifyHealth(OperationType.Subtraction, 5);
+            if (health.getCurrentHealth() <= 0)
```

Comment:
We talked about health for enemies can go negative, which gives us a reason to use the Addition operator (to bring up their health to zero). You should just check if it is zero.
```suggestion
            if (health.getCurrentHealth() == 0)
```